src/types.tsx

```diff
@@ -0,0 +1,46 @@
+export interface State {
```

Comment:
Do you mind changing this to AppState? I love that it's in a new file! Thanks for doing this. I think for dev empathy, we should be more specific with the name of State interface.